examples/xgboost/methods/guest/Cargo.toml

```diff
@@ -0,0 +1,11 @@
+[package]
+name = "xgboost"
+version = "0.1.0"
+edition = "2021"
+
+[workspace]
+
+[dependencies]
+risc0-zkvm = { path = "../../../../risc0/zkvm", default-features = false, features = ["std"] }
+forust-ml = "0.4.2"
+rmp-serde = "1.1.2"
```

Comment:
This can be dropped, we should always prefer to use the internal risc0 serde codec since it's optimized for the guest.

Reply:
The issue with passing the GradientBooster struct to the guest is that it causes the following error:
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: DeserializeBadBool', /Users/royrotstein/risczero/risc0/risc0/zkvm/src/guest/env.rs:415:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Guest panicked: called `Result::unwrap()` on an `Err` value: DeserializeBadBool', risc0/r0vm/src/lib.rs:186:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: failed to fill whole buffer', xgboost/src/main.rs:61:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```